File: backend/app/schemas/condition.py
"""Condition (diagnosis) document schema."""
from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from .enums import ConditionStatus
from .objectid import PyObjectId


def _utc_now() -> datetime:
    return datetime.now(timezone.utc)


class ConditionDoc(BaseModel):
    """Mongo document shape for conditions collection. Links to patient_id."""

    id: Optional[PyObjectId] = Field(None, alias="_id")
    patient_id: PyObjectId
    code: str
    display_name: Optional[str] = None
    status: ConditionStatus
    created_at: datetime = Field(default_factory=_utc_now)
    neo4j_synced_at: Optional[datetime] = None

    model_config = {"populate_by_name": True, "arbitrary_types_allowed": True}

File: express-backend/src/models/Condition.ts
import mongoose, { Schema, type Document, type Model } from 'mongoose';

export interface ConditionDocument extends Document {
  patient_id: mongoose.Types.ObjectId;
  code: string;
  display_name?: string;
  status: string;
  created_at: Date;
  neo4j_synced_at?: Date | null;
}

const ConditionSchema = new Schema<ConditionDocument>(
  {
    patient_id: { type: Schema.Types.ObjectId, ref: 'Patient', required: true },
    code: { type: String, required: true },
    display_name: { type: String },
    status: { type: String, required: true },
    created_at: { type: Date, default: () => new Date() },
    neo4j_synced_at: { type: Date, default: null },
  },
  {
    collection: 'conditions',
  },
);

export const Condition: Model<ConditionDocument> =
  mongoose.models.Condition || mongoose.model<ConditionDocument>('Condition', ConditionSchema);

